src/presentation/events-sender.service.ts

```diff
@@ -1,49 +1,70 @@
-import { Service } from 'typedi';
-import { Track } from '@domain';
 import { Context } from 'probot';
-
-const NextIssueText = link => `[Click here](${link}) for your next track`;
+import { Service } from 'typedi';
+import { Developer, DeveloperProgress, GetTracksUseCase, NextProgressUseCase, Track } from '@domain';
+import { RobotStrings } from './robot.strings';
 
 @Service()
 export class GithubEventSender {
-  private context: Context;
+  constructor(
+    private readonly getTracksUseCase: GetTracksUseCase,
+    private readonly nextProgressUseCase: NextProgressUseCase,
+  ) {}
 
-  async openEvent(context: Context, track: Track) {
-    this.context = context;
+  async openEvent(context: Context, developer: Developer) {
+    const tracks = await this.getTracksUseCase.exec();
+    const progress: DeveloperProgress = developer.progress;
+    let trackToSend: Track;
 
-    const isFirstTrackAndStep = track.number === 0 && track.steps.length === 1;
-    if (isFirstTrackAndStep) {
-      await this.createFirstIssue(track.title, track.steps[0].body);
+    const isNewUser: boolean = !progress;
+    if (isNewUser) {
+      context.log(`Creating first track for ${developer.name}...`);
+      trackToSend = tracks[0];
+      await this.createFirstIssue(context, trackToSend.title, trackToSend.steps[0].body);
+      return;
     }
 
-    const isNewTrack = track.steps.length === 1;
+    const finished: boolean = progress.completed === 1;
+    if (finished) {
+      context.log(`Congratulating ${developer.name} for finishing tutorial...`);
```

Comment:
Isso é executado alguma vez? Por que achei que se progress.completed === 1, na verdade cai no "nothing to do" do increment progress